pkg/cluster/statussqlfilter.go

```diff
@@ -0,0 +1,117 @@
+package cluster
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/kyma-incubator/reconciler/pkg/db"
+	"github.com/kyma-incubator/reconciler/pkg/model"
+)
+
+const (
+	CreatedAtColumnName       = "Created"
+	StatusCreatedAtColumnName = "StatusCreatedAt"
+	StatusColumnName          = "Status"
+	RuntimeIDColumnName       = "RuntimeID"
+	ConfigIDColumnName        = "ConfigID"
+)
+
+type statusSQLFilter interface {
+	Filter(dbType db.Type, colHdr *db.ColumnHandler) (string, error)
+}
+
+type statusFilter struct {
+	allowedStatuses []model.Status
+}
+
+func (sf *statusFilter) Filter(_ db.Type, statusColHdr *db.ColumnHandler) (string, error) {
+	statusColName, err := statusColHdr.ColumnName(StatusColumnName)
+	if err != nil {
+		return "", err
+	}
+	return fmt.Sprintf("%s IN ('%s')", statusColName, strings.Join(sf.statusesToStrings(), "','")), nil
+}
+
+func (sf *statusFilter) statusesToStrings() []string {
+	var result []string
+	for _, status := range sf.allowedStatuses {
+		result = append(result, string(status))
+	}
+	return result
+}
+
+type reconcileIntervalFilter struct {
+	reconcileInterval time.Duration
+}
+
+func (rif *reconcileIntervalFilter) Filter(dbType db.Type, statusColHdr *db.ColumnHandler) (string, error) {
+	statusColName, err := statusColHdr.ColumnName(StatusColumnName)
+	if err != nil {
+		return "", err
+	}
+	createdColName, err := statusColHdr.ColumnName(StatusCreatedAtColumnName)
+	if err != nil {
+		return "", err
+	}
+	switch dbType {
+	case db.Postgres:
+		return fmt.Sprintf(`%s IN ('%s', '%s', '%s') AND %s <= NOW() - INTERVAL '%.0f SECOND'`,
+			statusColName, model.ClusterStatusReady, model.ClusterStatusReconcileErrorRetryable, model.ClusterStatusDeleteErrorRetryable, createdColName, rif.reconcileInterval.Seconds()), nil
+	case db.SQLite:
```

Comment:
We are deprecating `SQLite`, and already stop to use it in any case, there is dedicate issue for cleanup the related code https://github.com/kyma-incubator/reconciler/issues/794.
This switch case we don't need to introduce.

Reply:
This is not introduced but instead is unified into one file. You can see we just moved this function to statussqlfilter.go from statusfilter.go. we didn't want to touch the logic because of this exact refactoring in #794, but we can also remove it now. What do you think?

Reply:
ok, sorry for this misunderstanding, then let's keep it.